pkg/framework/plugins/nodeutilization/highnodeutilization.go

```diff
@@ -19,40 +19,74 @@ package nodeutilization
 import (
 	"context"
 	"fmt"
-
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
-	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/klog/v2"
-
 	"sigs.k8s.io/descheduler/pkg/api"
 	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
 	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
+
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
 	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
 )
 
-// HighNodeUtilization evicts pods from under utilized nodes so that scheduler can schedule according to its strategy.
+const HighNodeUtilizationPluginName = "HighNodeUtilization"
+
+// HighNodeUtilization evicts pods from under utilized nodes so that scheduler can schedule according to its plugin.
 // Note that CPU/Memory requests are used to calculate nodes' utilization and not the actual resource usage.
-func HighNodeUtilization(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor, evictorFilter *evictions.EvictorFilter, getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc) {
-	if err := validateNodeUtilizationParams(strategy.Params); err != nil {
-		klog.ErrorS(err, "Invalid HighNodeUtilization parameters")
-		return
+
+type HighNodeUtilization struct {
+	handle     framework.Handle
+	args       *componentconfig.HighNodeUtilizationArgs
+	podFilter  func(pod *v1.Pod) bool
+	podEvictor *evictions.PodEvictor
+}
+
+var _ framework.Plugin = &HighNodeUtilization{}
+var _ framework.BalancePlugin = &HighNodeUtilization{}
+
+// NewHighNodeUtilization builds plugin from its arguments while passing a handle
+func NewHighNodeUtilization(args runtime.Object, handle framework.Handle, podEvictor *evictions.PodEvictor) (framework.Plugin, error) {
```

Comment:
All plugin constructors are expected to accept only `args` and `handle` parameters. Pod evictor is to be retrieved from the `handle` as `d.handle.Evictor()`. Is there a reason to the third `podEvictor` parameter?

Reply:
i will click resolve here, and discuss on below